### PR TITLE
Exclude tests in drivers

### PR DESCRIPTION
**Pull request by dpsutton**
Chop down on number of tests run in CI.

Exclude some tests from the drivers suite. Updates the invocation in `.github/actions/test-driver/action.yml` to be

```yaml
    - name: Test database driver
      run: clojure -X:dev:ci:ee:ee-dev:drivers:drivers-dev:test ${{ inputs.test-args }} :exclude-tags '[:mb/once]'
      shell: bash
```

In CI, 

Before: Running 3137 tests
After: Finding tests took 1.3 mins.: Running 2727 tests

There seems to be some jitter. Seems like things that hit a network or cloud can still take as long depending on the random run. Interesting results are in local dbs:

|Test     | Before | After  |
|---------|--------|--------|
|sqlite   |11m 46s | 8m 57s |
|postgres |17m 28s |11m 25s |
|sqlserver|13m 23s | 9m 56s |
|oracle   |14m  0s | 9m 45s |
|mysql    |37m 49s |26m 56s |
|mongo    |12m 20s | 8m 29s |
|maria    |17m 41s |13m 44s |
|druid    |12m 17s | 7m 41s |

Cloud DBs (i've seen some of these take longer in runs)

|Test     | Before | After  |
|---------|--------|--------|
|athena   | 35m 16s| 25m 6s |
|bigquery | 26m 35s| ---    |
|GA       | 11m 27s| 9m 33s |
|redshift | 29m 42s| 17m 3s |

```clojure
❯ clj -X:dev:test :exclude-tags '[:mb.test/once]'
WARNING: abs already refers to: #'clojure.core/abs in namespace: clojure.algo.generic.math-functions, being replaced by: #'clojure.algo.generic.math-functions/abs
operator.clj:170 recur arg for primitive local: sum is not matching primitive, had: Object, needed: long
Auto-boxing loop arg: sum
Warning: environ value /Users/dan/.sdkman/candidates/java/current for key :java-home has been overwritten with /Users/dan/.sdkman/candidates/java/17.0.1-zulu/zulu-17.jdk/Contents/Home
Looking for test namespaces in directory test
..instrumented #'metabase.util.malli/with-api-error-message
..instrumented #'metabase.util.honey-sql-2-extensions/identifier
..instrumented #'metabase.util.honey-sql-2-extensions/normalize-type-info
..instrumented #'metabase.util.honey-sql-2-extensions/with-database-type-info
..instrumented #'metabase.util.honey-sql-2-extensions/cast
..instrumented #'metabase.util.honey-sql-2-extensions/quoted-cast
..instrumented #'metabase.util.honey-sql-2-extensions/maybe-cast
WARNING: abs already refers to: #'clojure.core/abs in namespace: kixi.stats.math, being replaced by: #'kixi.stats.math/abs
WARNING: abs already refers to: #'clojure.core/abs in namespace: kixi.stats.test, being replaced by: #'kixi.stats.math/abs
WARNING: abs already refers to: #'clojure.core/abs in namespace: kixi.stats.distribution, being replaced by: #'kixi.stats.math/abs
Looking for test namespaces in directory shared/test
Skipping tests in `metabase.sample-data-test` due to :exclude-once
Skipping tests in `metabase.setup-test` due to :exclude-once
Skipping tests in `metabase.sync.analyze-test` due to :exclude-once
Skipping tests in `metabase.sync.field-values-test` due to :exclude-once
Skipping tests in `metabase.sync.sync-metadata.comments-test` due to :exclude-once
Skipping tests in `metabase.sync.sync-metadata.fields.sync-instances-test` due to :exclude-once
Skipping tests in `metabase.sync.sync-metadata.fields-test` due to :exclude-once
Skipping tests in `metabase.sync.sync-metadata.sync-database-type-test` due to :exclude-once
Skipping tests in `metabase.sync.sync-metadata.tables-test` due to :exclude-once
Skipping tests in `metabase.sync.sync-metadata-test` due to :exclude-once
Skipping tests in `metabase.sync.util-test` due to :exclude-once
Skipping tests in `metabase.sync-test` due to :exclude-once
Skipping tests in `metabase.task.follow-up-emails-test` due to :exclude-once
Skipping tests in `metabase.task.send-pulses-test` due to :exclude-once
Skipping tests in `metabase.task.task-history-cleanup-test` due to :exclude-once
Skipping tests in `metabase.task.upgrade-checks-test` due to :exclude-once
Skipping tests in `metabase.task-test` due to :exclude-once
Skipping tests in `metabase.util.cron-test` due to :exclude-once
Skipping tests in `metabase.util.date-2-test` due to :exclude-once
Skipping tests in `metabase.util.embed-test` due to :exclude-once
Skipping tests in `metabase.util.encryption-test` due to :exclude-once
Skipping tests in `metabase.util.files-test` due to :exclude-once
Skipping tests in `metabase.util.fonts-test` due to :exclude-once
Skipping tests in `metabase.util.honey-sql-1-extensions-test` due to :exclude-once
Skipping tests in `metabase.util.i18n.impl-test` due to :exclude-once
Skipping tests in `metabase.util.i18n.plural-test` due to :exclude-once
Skipping tests in `metabase.util.i18n-test` due to :exclude-once
Skipping tests in `metabase.util.malli.describe-test` due to :exclude-once
Skipping tests in `metabase.util.malli-test` due to :exclude-once
Skipping tests in `metabase.util.password-test` due to :exclude-once
Skipping tests in `metabase.util.regex-test` due to :exclude-once
Skipping tests in `metabase.util.retry-test` due to :exclude-once
Skipping tests in `metabase.util.schema-test` due to :exclude-once
Skipping tests in `metabase.util.ssh-test` due to :exclude-once
Skipping tests in `metabase.util-test` due to :exclude-once
Finding tests took 1.0 mins.
Running 2650 tests
```

Test namespaces can be annotated like:

```clojure
(ns ^:mb/once metabase.util.password-test
  (:require
   [clojure.test :refer :all]
   [metabase.test :as mt]
   [metabase.test.fixtures :as fixtures]
   [metabase.util.password :as u.password]))
```

And these can be excluded from running. In the future we can add things like `^:mb/long` to exclude long running tests from some CI checks in the future. `clj -X:dev:test :exclude-tags '[:mb/once :mb/long]'` can exclude them both.

Not implemented at this time are two features we might like in the future:
- a "positive" selector. Right now we have an `:only`. In the future we could have an `:only-tags` selector to run _only_ the tags selected.
- also this does not exclude individual tests with the tag. That can be some follow up work when we're ready. For this first pass just getting large swatchs of tests by namespace that are known to be insensitive to the particular driver.

### CODE REVIEW

File: test/metabase/test/data/impl.clj [hunk 1]
```diff
@@ -110,7 +110,7 @@
         (u/with-timeout sync-timeout-ms
           (let [reference-duration (or (some-> (get @reference-sync-durations database-name) u/format-nanoseconds)
                                        "NONE")
-                quick-sync? (not= database-name "test-data")]
+                quick-sync? ((complement #{"test-data" "sample-dataset"}) database-name)]
```
Comment (dpsutton) [inline test/metabase/test/data/impl.clj]: This should be the only strange thing in this PR. `metabase.query-processor.util.add-alias-info-test/join-source-query-join-test` tests a query shape and one of the fields has `:semantic_type :type/Category`. But the infra used to only run a full sync on "test-data". Which means some fields wouldn't have semantic_types unless some other test randomly did a full sync.

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj [hunk 2]
```diff
@@ -1,4 +1,4 @@
-(ns metabase-enterprise.advanced-permissions.api.group-manager-test
+(ns ^{:mb/tag :mb.test/once} metabase-enterprise.advanced-permissions.api.group-manager-test
```
Comment (camsaul) [inline enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj:1]: Could this work if we just did `^:mb.test/once`? It's a lot easier to type

File: test/metabase/test_runner.clj [hunk 3]
```diff
@@ -99,18 +101,22 @@
       ;; a namespace e.g. `metabase.whatever-test`
       (do
         (load-test-namespace symb)
-        (eftest.runner/find-tests symb)))))
+        (let [ns-tag (-> symb find-ns meta :mb/tag)]
+          (if (and (seq *exclude-tags*) ns-tag (contains? *exclude-tags* ns-tag))
+            (println (format "Skipping tests in `%s` due to :exclude-once" symb))
```
Comment (camsaul) [inline test/metabase/test_runner.clj:106]: I think you changed the name of this metadata?

**Review by camsaul — APPROVED** [review]
works for me, but I would vote for getting rid of `^:mb/tag` and just using `^:mb/once` directly. Then you can make `:exclude` just exclude any tests that have certain tags in the metadata

File: enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj (hunk 2 above)
Comment (dpsutton) [reply enterprise/backend/test/metabase_enterprise/advanced_permissions/api/group_manager_test.clj:1]: Yeah i can do that. Originally it was like this but in making it more like tags I figured this kept a single place in the meta to look. I'll switch it back to this style.

**Pull request merged**